js/filter.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+(function () {
+
+  var MAX_PINS_ON_MAP = 5;
+  var valueOfAllAds = 'any';
+
+  var typeFilterNode = document.querySelector('#housing-type');
+
+  var filterPinsOfType = function (dataElement) { // функция сравнения значения фильтра с типом жилья в объявлении
+    switch (typeFilterNode.value) {
+      case (dataElement.offer.type):
+        return true;
+      case (valueOfAllAds):
+        return true;
+      default:
+        return false;
+    }
+  };
```

Comment:
`сase` можно группировать. Элемент отфильтруется только если функция вернет `true`, а значит `default: return false` избыточен
```suggestion
  var filterPinsOfType = function (dataElement) { // функция сравнения значения фильтра с типом жилья в объявлении
    switch (typeFilterNode.value) {
      case (dataElement.offer.type):
      case (valueOfAllAds):
        return true;
    }
  };
```
Но в таком случае возникает вопрос. А нужен ли нам вообще `switch`?